modules/@apostrophecms/module/index.js

```diff
@@ -182,7 +182,10 @@ module.exports = {
                 res.header('Cache-Control', 'no-store');
               }
 
-              res.status(200);
+              // Keep "304 Not Modified" status if previously set
+              const statusCode = res.statusCode === 304 ? 304 : 200;
```

Comment:
express was still sending a `304` status code in the end (browser received 304), probably thanks to its work on ETags matching, but I think that would be better not to force a `200` here, when we're actually sending a `304` earlier (in [here](https://github.com/apostrophecms/apostrophe/pull/3692/files#diff-5e6c997e317e3d9d022272dd5b800a9ddd9965d767280831bd1148f202c1dc1dR186) and [there](https://github.com/apostrophecms/apostrophe/pull/3692/files#diff-b0584effe98676702427f4558a0ea60d8993af875d2879d2f44893e1b7f8aa58R223))